components/api-server/src/client.rs

```diff
@@ -0,0 +1,265 @@
+            };
+            let mongodb::bson::Bson::String(message) = msg else {
+                return Err(ClientError::MalformedData);
+            };
+            let value: serde_json::Value = serde_json::from_str(message)?;
+            Ok(value)
+        });
+
+        Ok(mapped)
+    }
+
+    /// Retrieves the current status of a previously submitted search job.
+    ///
+    /// # Arguments
+    ///
+    /// * `search_job_id` - The unique identifier (`u64`) of the search job whose status is being
+    ///   requested.
+    ///
+    /// # Returns
+    ///
+    /// A `Result` which is either:
+    ///
+    /// * `Ok(QueryJobStatus)`: The current status of the job, wrapped in the `QueryJobStatus` enum.
+    /// * `Err(ClientError)`: An error if the database query fails, the job ID is not found, or if
+    ///   the status value cannot be converted into a valid `QueryJobStatus`.
+    ///
+    /// # Errors
+    ///
+    /// This function can return a `ClientError` for reasons including:
+    /// * **Database Errors:** Failure to execute the SQL query (e.g., connection issues).
+    /// * **Row Not Found:** If no job with the given `search_job_id` exists.
+    /// * **Data Conversion Errors:** If the retrieved status integer is invalid or cannot be
+    ///   converted to `QueryJobStatus`.
+    pub async fn get_status(&self, search_job_id: u64) -> Result<QueryJobStatus, ClientError> {
+        let row = sqlx::query(&format!(
+            "SELECT status FROM `{QUERY_JOBS_TABLE_NAME}` WHERE id = ?"
+        ))
+        .bind(search_job_id)
+        .fetch_one(&self.sql_pool)
+        .await?;
+        let status: i32 = row.try_get("status")?;
+        Ok(QueryJobStatus::try_from(status)?)
+    }
+}
+
+#[derive(Clone, Serialize, Deserialize)]
+pub struct QueryConfig {
+    pub query_string: String,
+    #[serde(default)]
+    pub dataset: Option<String>,
+    #[serde(default)]
+    pub max_num_results: u32,
+    #[serde(default)]
+    pub begin_timestamp: Option<i64>,
+    #[serde(default)]
+    pub end_timestamp: Option<i64>,
+    #[serde(default)]
+    pub ignore_case: bool,
+}
+
+impl From<QueryConfig> for SearchJobConfig {
+    fn from(value: QueryConfig) -> Self {
+        Self {
+            dataset: value.dataset,
+            query_string: value.query_string,
+            max_num_results: value.max_num_results,
+            begin_timestamp: value.begin_timestamp,
+            end_timestamp: value.end_timestamp,
+            ignore_case: value.ignore_case,
+            ..Default::default()
+        }
+    }
+}
+
+#[derive(Error, Debug)]
+pub enum ClientError {
+    #[error("SQL error: {0}")]
+    Sql(#[from] sqlx::Error),
+
+    #[error("MongoDB error: {0}")]
+    Mongo(#[from] mongodb::error::Error),
+
+    #[error("Query is not succeeded")]
+    QueryNotSucceeded,
+
+    #[error("IO error")]
+    Io(#[from] std::io::Error),
+
+    #[error("Malformed database")]
+    MalformedData,
+}
```

Comment:
```suggestion
#[derive(Error, Debug)]
pub enum ClientError {
    #[error("`sqlx::Error`: {0}")]
    Sql(#[from] sqlx::Error),

    #[error("`mongodb::error::Error`: {0}")]
    Mongo(#[from] mongodb::error::Error),

    #[error("Query is failed")]
    QueryFailed,

    #[error("`std::io::Error`: {0}")]
    Io(#[from] std::io::Error),

    #[error("Malformed data")]
    MalformedData,
}
```
- To be consistent with `clp-rust-utils` in terms of error definition.
- "query not succeeded" to "query failed" as it's shorter and simplier?
- IO error needs `{0}`? Or u omitted it because the message will always be the same?
- "database" -> "data"